### PR TITLE
PLT-143: Enable and configure dependabot for security updates across all repos

### PR DESCRIPTION
**Pull request by gfreeman-navapbc**
## 🎫 Ticket

https://jira.cms.gov/browse/PLT-143

## 🛠 Changes

Add Dependabot config

## ℹ️ Context for reviewers

We decided to converge on Dependabot for our dependency management solution. We'll only be alerting on critical security updates.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

**Review by kyeah — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+version: 2
+updates:
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    open-pull-requests-limit: 0
+
+  - package-ecosystem: "terraform"
```
Comment (gsf) [inline .github/dependabot.yml:13]: Do we need terraform when there's none in the repo?

Comment (gfreeman-navapbc) [reply .github/dependabot.yml:13]: Probably not, was mostly using the same config for all BCDA repos, I'll go back through and remove anything that doesn't quite fit

**Review by gsf — DISMISSED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# To get started with Dependabot version updates, you'll need to specify which
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+version: 2
+updates:
+  - package-ecosystem: "gomod"
```
Comment (gsf) [inline .github/dependabot.yml:7]: Looks like nodejs and ruby are used in this repo

Comment (gsf) [reply .github/dependabot.yml:7]: Sorry I wasn't clear there. We can also drop gomod

**Review by gsf — APPROVED** [review]

**Pull request merged**